internal/rooms/biomes.go

```
@@ -1,168 +1,248 @@
-			symbol:      '~',
-			description: `Shores are the transition between land and water. You can usually fish from them.`,
-		},
-		`water`: {
-			name:           `Deep Water`,
-			symbol:         '≈',
-			description:    `Deep water is dangerous and usually requires some sort of assistance to cross.`,
-			requiredItemId: 20030,
-		},
-		`forest`: {
-			name:        `Forest`,
-			symbol:      '♣',
-			description: `Forests are wild areas full of trees. Animals and monsters often live here.`,
-			burns:       true,
-		},
-		`mountains`: {
-			name:        `Mountains`,
-			symbol:      '⩕', //'▲',
-			description: `Mountains are difficult to traverse, with roads that don't often follow a straight line.`,
-		},
-		`cliffs`: {
-			name:        `Cliffs`,
-			symbol:      '▼',
-			description: `Cliffs are steep, rocky areas that are difficult to traverse. They can be climbed up or down with the right skills and equipment.`,
-		},
-		`swamp`: {
-			name:        `Swamp`,
-			symbol:      '♨',
-			darkArea:    true,
-			description: `Swamps are wet, muddy areas that are difficult to traverse.`,
-		},
-		`snow`: {
-			name:        `Snow`,
-			symbol:      '❄',
-			description: `Snow is cold and wet. It can be difficult to traverse, but is usually safe.`,
-		},
-		`spiderweb`: {
-			name:        `Spiderweb`,
-			symbol:      '🕸',
-			darkArea:    true,
-			description: `Spiderwebs are usually found where larger spiders live. They are very dangerous areas.`,
-		},
-		`cave`: {
-			name:        `Cave`,
-			symbol:      '⌬',
-			darkArea:    true,
-			description: `The land is covered in caves of all sorts. You never know what you'll find in them.`,
-		},
-		`desert`: {
-			name:        `Desert`,
-			symbol:      '*',
-			description: `The harsh desert is unforgiving and dry.`,
-		},
-		`farmland`: {
-			name:        `Farmland`,
-			symbol:      ',',
-			description: `Wheat or other food is grown here.`,
-			burns:       true,
+func loadHardcodedBiomes() {
```

Curious - why are these loaded this way? It seems we should have a single default biome that gets loaded incase biomes are missing, but why city, land, etc.? Shouldn't these just be expected to be defined in the yaml?

---

Okay, re-reading the notes, I see the intention is "Backwards Compatibility: Falls back to hardcoded defaults if no biome files are found"

I think we can omit it for now. Lets assume they will have them, or else all biomes will failover to a singularly defined default.

This will be an easy problem to identify/fix if someone is on an old version and only updated the binary.

Hopefully by `v1.0.0` we will have eliminated hard coded data and moved it all to externally loaded yaml files.

---

Check. I'll set up a single fallback biome. I set all the hard coded ones up to prevent the default world from creating issues, while still supporting all the biomes in that world in case someone wanted to create their own rooms.

I had multiple new biomes in my own world, and all of this is actually a result of me being tired of switching branches back and forth in my fork when I wanted to test my own world vs. working on the main codebase.